### PR TITLE
Avoid running into timeouts when dumping shoot state

### PR DESCRIPTION
**Pull request by timebertt**
Co-authored-by: Tim Schrodi <tim.schrodi@sap.com>

**How to categorize this PR?**

/area testing
/kind enhancement
/priority normal

**What this PR does / why we need it**:
We see some integration test runs, where we run into timeouts when dumping the shoot's state, because the API server is not running.
With this PR we check if the API server is running before dumping, so we can avoid running into timeouts, so that the Seed state will still be dumped.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: test/framework/shootframework.go [hunk 1]
```diff
@@ -322,3 +325,19 @@ func (f *ShootFramework) WaitForShootCondition(ctx context.Context, interval, ti
 		return retry.MinorError(fmt.Errorf("shoot %q does not yet have expected condition", shoot.Name))
 	})
 }
+
+func (f *ShootFramework) IsAPIServerRunning(ctx context.Context) (bool, error) {
```
Comment (rfranzke) [inline test/framework/shootframework.go:331]: doc string

**Review by rfranzke — APPROVED** [review]

File: test/framework/shoot_utils.go [hunk 2]
```diff
@@ -108,6 +109,12 @@ func (f *ShootFramework) DumpState(ctx context.Context) {
 			if err := f.dumpNodes(ctx, ctxIdentifier, f.ShootClient); err != nil {
 				f.Logger.Errorf("unable to dump information of nodes from shoot %s: %s", f.Shoot.Name, err.Error())
 			}
+		} else {
+			errMsg := ""
+			if err != nil {
+				errMsg = ": " + err.Error()
+			}
+			f.Logger.Errorf("unable to dump resources from shoot %s: API server is currently not running%s", f.Shoot.Name, errMsg)
```
Comment (dguendisch) [inline test/framework/shoot_utils.go:117]:
```suggestion
			f.Logger.Errorf("unable to dump resources from shoot %s: API server is currently not running %s", f.Shoot.Name, errMsg)
```

**Review by dguendisch — CHANGES_REQUESTED** [review]
Good idea, thanks for spotting. Only a nit comment.

File: test/framework/shoot_utils.go (hunk 2 above)
Comment (timebertt) [reply test/framework/shoot_utils.go:117]: Hmm, `errMsg` is either empty or filled with `: <some error message>`...

Comment (dguendisch) [reply test/framework/shoot_utils.go:117]: oops, you're right, thanks!

**Review by dguendisch — APPROVED** [review]

**Pull request merged**